app/models/health/engagement_calculator.rb

```diff
@@ -0,0 +1,84 @@
+
+      if ssm_form.completed?
+        return [:too_old, 'Self-Sufficiency Matrix', new_client_health_self_sufficiency_matrix_form_path(@client), false, "Last completed on #{ssm_form.completed_at.to_date}"] if ssm_form.completed_at < @valid_after
+        return [:expired, 'Self-Sufficiency Matrix', new_client_health_self_sufficiency_matrix_form_path(@client), false, "Last completed on #{ssm_form.completed_at.to_date}"] unless ssm_form.active?
+
+        [:valid, 'Self-Sufficiency Matrix', client_health_self_sufficiency_matrix_form_path(@client, ssm_form), true, nil]
+      else
+        return [:too_old, 'Self-Sufficiency Matrix', new_client_health_self_sufficiency_matrix_form_path(@client), false, "Last completed on #{ssm_form.completed_at.to_date}"] if ssm_form.created_at < @valid_after
+
+        prior_forms = @patient.self_sufficiency_matrix_forms.count > 1
+        return [:being_updated, 'Self-Sufficiency Matrix', client_health_self_sufficiency_matrix_form_path(@client, ssm_form), true, "Started on #{ssm_form.created_at.to_date}"] if prior_forms
+
+        [:in_progress, 'Self-Sufficiency Matrix', client_health_self_sufficiency_matrix_form_path(@client, ssm_form), true, "Started on #{ssm_form.created_at.to_date}"]
+      end
+    end
+
+    def cha_status
+      cha_form = @patient.comprehensive_health_assessments.recent.first
+      return [:no_signed_form, 'Comprehensive Health Assessment', new_client_health_cha_path(@client), false, nil] if cha_form.blank?
+      return [:too_old, 'Comprehensive Health Assessment', new_client_health_cha_path(@client), false, "Last completed on #{cha_form.completed_at.to_date}"] if cha_form.completed_at < @valid_after
+
+      if cha_form.completed?
+        return [:expired, 'Comprehensive Health Assessment', new_client_health_cha_path(@client), false, "Last completed on #{cha_form.completed_at.to_date}"] unless cha_form.active?
+
+        [:valid, 'Comprehensive Health Assessment', client_health_cha_path(@client, cha_form), true, nil]
+      else
+        prior_forms = @patient.comprehensive_health_assessments.count > 1
+        return [:being_updated, 'Comprehensive Health Assessment', client_health_cha_path(@client, cha_form), true, "Started on #{cha_form.created_at.to_date}"] if prior_forms
+
+        [:in_progress, 'Comprehensive Health Assessment', client_health_cha_path(@client, cha_form), true, "Started on #{cha_form.created_at.to_date}"]
+      end
+    end
+
+    def careplan_status
+      careplan = @patient.careplans.recent.first
+      return [:no_signed_form, 'PCTP Signed', new_client_health_careplan_path(@client), false, nil] if careplan.blank?
+      return [:valid, 'PCTP Signed', client_health_careplan_path(@client, careplan), false, nil] if careplan.active?
+
+      if careplan.editable?
+        return [:being_updated, 'PCTP Signed', client_health_careplan_path(@client, careplan), false, "Started on #{careplan.created_at.to_date}"] if @patient.careplans.expired.exists?
```

Comment:
Is this the right check? `@patient.careplans.expired.exists?`?  Expired doesn't seem right.

Reply:
If the current careplan is editable, and there exists an expired one, we must be updating the careplan, otherwise the initial careplan is in progress. It is possible that they could open a careplan, and never complete it for a year, and then it would swap from in progress to being updated, but that seemed like an unnecessary corner case, and the other way we have identified renewals was when there was more than one of the form on the client, and that doesn't look like a clean signal for careplans (they seem to produce multiple careplan documents sometimes).